Add genre metadata support for streaming services

RobStanfield opened the pull request:
## Summary

- Add genre extraction for Qobuz and Spotify tracks during catalog refresh
- Add "Fetch metadata from service" context menu for existing playlist items
- Enable inline tag editing (genre, etc.) for streaming service tracks
- Enable multi-track "Set value to" for streaming service tracks
- Add request throttling for metadata fetching to avoid rate limiting

## Details

This PR adds comprehensive genre metadata support for streaming services (Qobuz, Spotify, Tidal, Subsonic):

**Genre extraction during catalog refresh:**
- Qobuz: Extracts genre from `album.genre.name` in the API response
- Spotify: Extracts genre from artist's genres array (requires additional API call)

**Fetch metadata for existing tracks:**
- New right-click menu option "Fetch metadata from service" 
- Works on selected Qobuz/Spotify tracks that don't have genre
- Throttled queue (200ms between requests) to avoid rate limiting

**Inline editing support:**
- `Song::IsEditable()` now returns true for streaming services
- `Playlist::setData()` handles streaming tracks (saves to playlist DB)
- `MainWindow::SelectionSetValue()` handles streaming tracks for multi-select
- `EditTagDialog` properly saves metadata changes for streaming tracks

## Test plan

- [ ] Add Qobuz/Spotify favorites and verify genre column is populated
- [ ] Right-click existing tracks without genre → "Fetch metadata from service"
- [ ] Verify genre appears after fetching
- [ ] Right-click genre column → "Edit tag genre" → change value → verify it persists
- [ ] Select multiple tracks → right-click → "Set genre to 'X'" → verify all updated
- [ ] Restart Strawberry and verify edited genres persist

RobStanfield committed:
Get genre metadata for Tidal, Qobuz and Spotify

Extract genre information when fetching favorites and search results.
Genre is now populated in the collection and playlists for
tracks from these streaming services.

RobStanfield committed:
Fetch metadata and allow editing for stream songs

jonaski commented on src/core/mainwindow.cpp:
```diff
@@ -2214,6 +2234,142 @@ void MainWindow::RescanSongs() {
+
+#ifdef HAVE_QOBUZ
+    if (source == Song::Source::Qobuz) {
+      track_id = song.url().path();
```
Use `song.song_id()`

jonaski commented on src/core/mainwindow.cpp:
```diff
@@ -2214,6 +2234,142 @@ void MainWindow::RescanSongs() {
+
+#ifdef HAVE_SPOTIFY
+    if (source == Song::Source::Spotify) {
+      QString url_str = song.url().toString();
```
Use `song.song_id()`

jonaski reviewed (changes requested):
I don't mind adding support for editing metadata, but I think the "Fetch metadata from service" should update the entire metadata, not just genre

RobStanfield commented:
Addressed the feedback:

1. **Fetch all metadata, not just genre** - `MetadataRequest` classes now return full `Song` objects with title, artist, album, albumartist, genre, track, disc, year, duration, and cover art. The callback in `MainWindow` applies all non-empty fields to the playlist item.

2. **Use `song_id()`** - Implemented with URL fallback since `song_id()` is empty when loaded from playlist (not persisted). Code tries `song_id()` first, falls back to URL parsing if empty.

Tested both Qobuz and Spotify - metadata fetch working correctly for multiple tracks.

xstable commented:
May I ask a question here - sorry, if it's may be off-topic.. but it's regarding to genre fetching. 
If I currently fetch Metadata, I never get any genre. Should I get the genre, or is there any way to enable this?
I've currently spotify (free version) connected.
